src/lib/DataTable.tsx

```diff
@@ -87,20 +87,40 @@ interface TableRowProps<T> {
     headerData?: boolean;
     onClick?: (row: T, index: number, event: React.MouseEvent<HTMLTableRowElement>) => void;
     span?: {[colName: string]: number};
+    getColSpansOfRow?: (
+        p: Omit<TableRowProps<T>, 'getColSpansOfRow' | 'onClick' | 'className'>,
+    ) => {[colName: string]: number} | undefined;
 }
 
 class TableRow<T> extends React.PureComponent<TableRowProps<T>> {
     static defaultProps = {
         footer: false,
     };
     render() {
-        const {className, columns, row, index, odd, footer, span, headerData} = this.props;
+        const {className, columns, row, index, odd, footer, span, getColSpansOfRow, headerData} =
+            this.props;
+        const colSpans = getColSpansOfRow ? getColSpansOfRow(this.props) : undefined;
+
+        let cols = [];
+
+        if (colSpans) {
+            for (let i = 0; i <= columns.length - 1; ++i) {
+                cols.push(columns[i]);
```

Comment:
It seems like a bad idea to refile the array for every row, cause it brings additional performance slowdown for large tables without virtualization. It is better to use additional numeric variable, for example `restColSpan` from last defined colSpan value and decrement it. Something like:
```ts
...
let restColSpan = 0;
{columns.map((...) => {
    if (--restColSpan > 0) {
         return null;
    }
    ...
})}
```